### ISSUE TITLE
Conflicts not shown

### ISSUE THREAD

**Issue by Leka1B** [kevinmkchin/Obsidian-GitHub-Sync#13]
Whenever i have a conflicting merge, the conflicting files are not described. Instead, i see only "Merge conflicts in:   ".
As I understand, I should instead see the names of the files or links to them? If not, feel free to close this issue. 

If I can successfully reproduce a conflict i will post a screenshot

**Comment by Leka1B** [comment]
Here is a screenshot as promised:
![image](https://github.com/user-attachments/assets/cdd3261e-84dd-470a-8332-12c1020ca995)

**Comment by kevinmkchin** [comment]
Yes the pop up should list the name of each file that is conflicting. It should also open each of the conflicting files. It worked before but I may have broken it at some point. I won't have time to look into this soon but feel free to take a look yourself. 

You'll have to look at the [simple-git](https://www.npmjs.com/package/simple-git) package that this plugin is using. Making and testing changes can be pretty annoying, and also that package doesn't have the best documentation and its a bit of a nightmare but you might have to dive into their [code](https://github.com/steveukx/git-js/tree/main/simple-git).

I'm also incorrectly assuming that any exceptions thrown during `git pull` is a merge conflict, but that's not always the case. Ideally we would check, but I didn't bother.

https://github.com/kevinmkchin/Obsidian-GitHub-Sync/blob/dacf86bb39f433a7a34151e8dc8a0030a7bc9460/main.ts#L104

**Comment by Leka1B** [comment]
Thanks for providint the links to their github. I decided to just print the whole status and hope that the Notice can handle whatever amounts of text it outputs...

**Comment by PittBie** [comment]
![image](https://github.com/user-attachments/assets/eac88522-8ee8-4c42-862c-51f980501921)

I have the same issue. Upon checking GitHub Desktop it says that the current branch `master` hasn't been published to the remote yet.
![image](https://github.com/user-attachments/assets/e6bbefd7-a2b3-48fa-b603-2d2b93c0b8f1)
If I then publish this branch the commits and changes are successfully pushed.

How to reproduce:
- I have a wiki setup in a private repo.
- I am using the https: URL and already have the repo properly setup, so that it works via GitHub Desktop

**Comment by Leka1B** [comment]
I have a suspicion it's caused by syncing files that store currently open windows and such. Try editing gitignore so that only the actual markdown files (i also sync extensions). Hope this helps

**Comment by fitzage** [comment]
I’ve removed the entire .obsidian folder from the repo, so all I have are markdown files and attachments. I get this merge conflict error on every sync, without any files listed, because there aren’t any actual conflicts. The changes get committed to git just fine, but they never get pushed. I have to manually push the changes.

**Comment by fitzage** [comment]
Ah, this is because the default branch has to be `main` instead of `master`. Once I changed to main, the erroneous error went away, and the actual sync with the remote happened successfully.
